File: src/workflows/objectiveEngine.ts
/**
 * Converted from Python: core/workflows/objective_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

var OBJECTIVE_STEPS: any = {"extract_dashboard": ["navigate_dashboard", "capture_widgets", "capture_tables"], "extract_invoices": ["open_invoices", "paginate", "extract_rows"], "monitor_metrics": ["open_dashboard", "observe_metrics", "checkpoint"], "capture_notifications": ["observe_notifications", "capture_events", "checkpoint"], "extract_repository": ["scan_repository", "extract_structure", "compile_graph"], "monitor_runtime": ["observe_runtime", "capture_mutations", "checkpoint"], "capture_terminal": ["open_terminal", "capture_output", "checkpoint"], "monitor_application": ["observe_application", "capture_state", "checkpoint"], "extract_api_surface": ["discover_routes", "extract_endpoints", "compile_api"], "monitor_infrastructure": ["observe_infra", "capture_status", "checkpoint"]};
export function buildRuntimeObjective(objective: any, priority: any = 0): any {
  var steps: any = [...py.iter(py.get(OBJECTIVE_STEPS, objective, ["observe", "extract", "checkpoint"]))];
  return {"objective": objective, "priority": priority, "steps": steps, "bounded": true};
}

File: lib/src/workflows/objective_engine.dart
// Port of core/workflows/objective_engine.py

const Map<String, List<String>> objectiveSteps = <String, List<String>>{
  'extract_dashboard': <String>[
    'navigate_dashboard',
    'capture_widgets',
    'capture_tables'
  ],
  'extract_invoices': <String>['open_invoices', 'paginate', 'extract_rows'],
  'monitor_metrics': <String>[
    'open_dashboard',
    'observe_metrics',
    'checkpoint'
  ],
  'capture_notifications': <String>[
    'observe_notifications',
    'capture_events',
    'checkpoint'
  ],
  'extract_repository': <String>[
    'scan_repository',
    'extract_structure',
    'compile_graph'
  ],
  'monitor_runtime': <String>[
    'observe_runtime',
    'capture_mutations',
    'checkpoint'
  ],
  'capture_terminal': <String>['open_terminal', 'capture_output', 'checkpoint'],
  'monitor_application': <String>[
    'observe_application',
    'capture_state',
    'checkpoint'
  ],
  'extract_api_surface': <String>[
    'discover_routes',
    'extract_endpoints',
    'compile_api'
  ],
  'monitor_infrastructure': <String>[
    'observe_infra',
    'capture_status',
    'checkpoint'
  ],
};

Map<String, dynamic> buildRuntimeObjective(
  String objective, [
  int priority = 0,
]) {
  final List<String> steps = List<String>.from(
    objectiveSteps[objective] ?? <String>['observe', 'extract', 'checkpoint'],
  );

  return <String, dynamic>{
    'objective': objective,
    'priority': priority,
    'steps': steps,
    'bounded': true,
  };
}
